### DISCUSSION TITLE
Swell with Astro

### DISCUSSION THREAD

**Discussion by tonyoconnell** [swellstores/community#120] [Help] [answered]
Hello. I saw a link to Astro as a supported Jamstack framework on the Swell home page. That's wonderful. Does anyone know of any projects, example code or tutorials about using Swell with Astro?

**Comment by maarten2424** [comment, marked as answer]
Hey @tonyoconnell thanks for reaching out!
We don't have a starter-kit or example of using Swell with Astro _yet_, but integration of Swell with Astro (or any other framework) is still possible and fairly easy!
Example:

1. Create a Astro project with Astro Create:
```shell
# create a new project with npm
npm create astro@latest
```

2. Create a `utils/swell.js` file to include Swell-js
```javascript
import swell from 'swell-js'

// Initialize the client first
swell.init('store-id', 'secret-key')

// Export the swell instance
export default swell
```

3. Fetch your products / data from Swell!
```javascript
// src/pages/index.astro

---
import BaseLayout from '../layouts/BaseLayout.astro'
import swell from '../../utils/swell.js'
 
// Now you can use any method from swell-js
const products = await swell.products.list({
  category: 't-shirts',
  limit: 25,
  page: 1
})
---
 
<BaseLayout products={products} />

```

Please stay in the loop, because we are launching `swell-sdk` soon, which makes implementation even more easier!

**Comment by nonfungibletunji** [comment]
any updates to swell astro integration
